docs/it/docs/python-types.md

```diff
@@ -0,0 +1,314 @@
+
+**FastAPI** è interamente basato sui type hints per i numerosi vantaggi che essi offrono.
+
+Ma anche se non hai mai usato o mai userai **FastAPI**, ti sarà utile capire cosa siano.
+
+!!! nota
+    Se sei un esperto di Python, e quindi saprai tutto sui type hints, puoi passare al capitolo successivo.
+
+## Motivazione
+
+Iniziamo con un semplice esempio:
+
+```Python
+{!../../../docs_src/python_types/tutorial001.py!}
+```
+
+Il risultato di questo programma è il seguente:
+
+```
+John Doe
+```
+
+La funzione fa questo:
+
+* Prende `first_name` e `last_name`.
+* Converte la prima lettera di entrambi in maiuscolo con `title()`.
+* Li <abbr title="Li unisce in una singola entità. Inserisce prima i contenuti della prima, poi dell'altra. In inglese: concatenate">concatena</abbr> inserendo uno spazio in mezzo.
+
+```Python hl_lines="2"
+{!../../../docs_src/python_types/tutorial001.py!}
+```
+
+### Modifichiamolo
+
+È un programma molto semplice.
+
+Ma adesso immagina che tu lo stia scrivendo da zero.
+
+A un certo punto inizi a definire la funzione e i parametri...
+
+Però poi ti serve chiamare "quel metodo che trasforma la prima lettera in maiuscolo".
+
+È `upper`? Oppure `uppercase`? `first_uppercase`? `capitalize`?
+
+Quindi provi a consultare il caro amico di ogni programmatore: l'autocompletamento dell'editor di testo.
+
+Digiti il primo parametro della funzione, `first_name`, poi un punto (`.`) e premi `Ctrl+Spazio` per innescare il completamento automatico.
+
+Purtroppo, non compare nulla di utile:
+
+<img src="https://fastapi.tiangolo.com/img/python-types/image01.png">
+
+### Aggiungi i tipi
+
+Modifichiamo l'ultimo programma aggiungendo le annotazioni di tipo.
+
+Cambieremo esattamente questa parte, i parametri della funzione, da:
+
+```Python
+    first_name, last_name
+```
+
+a:
+
+```Python
+    first_name: str, last_name: str
+```
+
+Solo questo.
+
+Questi sono i "type hints":
+
+```Python hl_lines="1"
+{!../../../docs_src/python_types/tutorial002.py!}
+```
+
+Attenzione che non è la stessa cosa di dichiarare i valori di default dei parametri della funzione:
+
+```Python
+    first_name="john", last_name="doe"
+```
+
+È una cosa diversa.
+
+Usiamo i due puntini (`:`), non l'uguale (`=`).
+
+E aggiungere le annotazioni di tipo normalmente non cambia il comportamento di un programma.
+
+Adesso immagina che stai ancora creando quella funzione, ma ora con i type hints.
+
+Allo stesso punto di prima, provi ad attivare l'autocompletamento con `Ctrl+Spazio` e vedi:
+
+<img src="https://fastapi.tiangolo.com/img/python-types/image02.png">
+
+In questo modo puoi vedere la lista delle opzioni fino a quando trovi quella che sembra più giusta:
+
+<img src="https://fastapi.tiangolo.com/img/python-types/image03.png">
+
+## Altri vantaggi
+
+Osserva questa funzione, ha già le annotazioni di tipo:
```

Comment:
It seems that something is missing, but I'm not sure. It's been a while since I last spoke Italian. Does my suggestion make sense?
```suggestion
Osserva questa funzione, puoi vedere che ha già le annotazioni di tipo:
```

Reply:
Grazie, e complimenti per il tuo italiano 🙂